packages/nimble-components/src/table/specs/table-column-placeholder-hld.md

```diff
@@ -0,0 +1,151 @@
+# Placeholders for table columns HLD
+
+## Problem Statement
+
+In some cases, an application may want to display a placeholder value in the table when a record does not have a value for a given field. The application should be able to customize that placeholder based on their needs.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Nimble issue 1538](https://github.com/ni/nimble/issues/1538)
+-   [Nimble issue 1511](https://github.com/ni/nimble/issues/1511)
+
+## Implementation / Design
+
+### High-level behavior
+
+There are two places where a value is displayed in the table that placeholders need to be considered: (1) in a cell and (2) in a group row. The cell placeholder will be configurable on each column that supports having a placeholder, and it will default to an empty string. The group row placeholder will come from the table's localization provider, and it will not be configurable through a column's API.
+
+Placeholders within a cell will be rendered with nimble's placeholder font, which is currently 60% opacity of nimble's body font.
+
+Placeholders within a group row will have no special visual treatment.
```

Comment:
I think this is a question for @fredvisser and @NIbokeefe -- should a group row placeholder have special visual treatment to avoid ambiguity between "real" data and our placeholder?

As a note, I pulled the current statement from the requirements in #1538, specifically [this comment](https://github.com/ni/nimble/issues/1538#issuecomment-1887951554):
>Should placeholder text be dimmed? Only in cells, but not group headers?
Yes, placeholder text should be dimmed (standard placeholder token value) in the data rows, but not in group headers.